cvat/apps/engine/task.py

```diff
@@ -960,206 +939,317 @@ def update_progress(progress):
+                            "Can't find image '{}' width or height info in the manifest"
+                            .format(f"{image_info['name']}{image_info['extension']}")
+                        )
 
-        futures = queue.Queue(maxsize=settings.CVAT_CONCURRENT_CHUNK_PROCESSING)
-        with concurrent.futures.ThreadPoolExecutor(max_workers=2*settings.CVAT_CONCURRENT_CHUNK_PROCESSING) as executor:
-            for chunk_idx, chunk_data in generator:
-                db_data.size += len(chunk_data)
-                if futures.full():
-                    process_results(futures.get().result())
-                futures.put(executor.submit(save_chunks, executor, chunk_idx, chunk_data))
+                if not image_size:
+                    image_size = extractor.get_image_size(frame_id)
 
-            while not futures.empty():
-                process_results(futures.get().result())
+                images.append(
+                    models.Image(
+                        data=db_data,
+                        path=os.path.relpath(image_path, upload_dir),
+                        frame=frame_id,
+                        width=image_size[0],
+                        height=image_size[1],
+                    )
+                )
 
     if db_task.mode == 'annotation':
-        models.Image.objects.bulk_create(db_images)
-        created_images = models.Image.objects.filter(data_id=db_data.id)
+        models.Image.objects.bulk_create(images)
+        images = models.Image.objects.filter(data_id=db_data.id)
 
         db_related_files = [
             models.RelatedFile(data=image.data, primary_image=image, path=os.path.join(upload_dir, related_file_path))
-            for image in created_images
+            for image in images
             for related_file_path in related_images.get(image.path, [])
         ]
         models.RelatedFile.objects.bulk_create(db_related_files)
-        db_images = []
     else:
         models.Video.objects.create(
             data=db_data,
             path=os.path.relpath(video_path, upload_dir),
-            width=video_size[0], height=video_size[1])
+            width=video_frame_size[0], height=video_frame_size[1]
+        )
 
+    # validate stop_frame
     if db_data.stop_frame == 0:
         db_data.stop_frame = db_data.start_frame + (db_data.size - 1) * db_data.get_frame_step()
     else:
-        # validate stop_frame
         db_data.stop_frame = min(db_data.stop_frame, \
             db_data.start_frame + (db_data.size - 1) * db_data.get_frame_step())
 
     slogger.glob.info("Found frames {} for Data #{}".format(db_data.size, db_data.id))
-    _save_task_to_db(db_task, job_file_mapping=job_file_mapping)
+    _create_segments_and_jobs(db_task, job_file_mapping=job_file_mapping)
+
+    if (
+        settings.MEDIA_CACHE_ALLOW_STATIC_CACHE and
+        db_data.storage_method == models.StorageMethodChoice.FILE_SYSTEM
+    ):
+        _create_static_chunks(db_task, media_extractor=extractor)
```

Comment:
Is it expected behavior?
- when there was a `manifest.jsonl` file uploaded together with images, `use_cache` == False, `MEDIA_CACHE_ALLOW_STATIC_CACHE` == True -> task created successfully with static chunks, but before that manifest index was initialized. That means that despite use_cache==False and MEDIA_CACHE_ALLOW_STATIC_CACHE==True values, we prepare manifest (if it was not passed), initialize a manifest index, and then prepare static chunks. It seems odd.

If I remember correctly, previously it was not allowed to pass a manifest file when use_cache == False.

Reply:
Yes, I changed it to generate manifest in any case. Basically, the idea is to use it for optimized data access on the server side regardless of the task creation method. It's also possible to use it even if the task was created with static chunks, as the server config may change.